src/Language/GraphQL/Draft/Syntax.hs

```diff
@@ -133,12 +174,31 @@ parseName text = maybe (fail errorMessage) pure $ mkName text
   where
     errorMessage = T.unpack text <> " is not valid GraphQL name"
 
+parseSuffix :: MonadFail m => Text -> m NameSuffix
+parseSuffix text = maybe (fail errorMessage) pure $ mkNameSuffix text
+  where
+    errorMessage = T.unpack text <> " is not valid GraphQL suffix"
+
 -- | Construct a 'Name' value at compile-time.
 litName :: Text -> SpliceQ Name
 litName txt = liftSplice do
   name <- parseName txt
   examineSplice [||name||]
 
+-- | Construct a 'NameSuffix' value at compile-time.
+litSuffix :: Text -> SpliceQ NameSuffix
+litSuffix txt = liftSplice do
+  name <- parseSuffix txt
+  examineSplice [||name||]
```

Comment:
it would probably be useful to have quasiquoters for this & `litGQLIdentifier` but that could certainly be done in a follow-up PR.